### DISCUSSION TITLE
Dynamically resizing a TextBox based on the inserted text's height.

### DISCUSSION THREAD

**Discussion by acidicly** [TomSchimansky/CustomTkinter#1345] [Q&A]
I am looking for help, is there a setting that can tell the textbox to just fill the height based on the inserted text's height.

**Comment by jh-jensen** [comment]
I'm trying to do something similar. Closest I've got is to count the number of lines in the textbox with:
`lines = e.widget.tk.call((e.widget, "count", "-update", "-displaylines", "1.0", "end"))`
And then adjusting the height with configure (fx `e.widget.master.configure(height=lines*30+12)),` but I have some weird problem with some extra height being added, and the extra height added dynamically changes based on how many lines there are. I spent so much time trying to figure out why this happens but still don't know. 
If anyone got it to work, please update us.

**Comment by MaxNiark** [comment]
please use def from scaling factor from tk class object.

 written in complexe example : 

![image](https://user-images.githubusercontent.com/126392328/226095887-e6618691-b437-454d-820e-64c40d2eb8b0.png)

**Reply by MaxNiark** [reply]
![image](https://user-images.githubusercontent.com/126392328/226095977-9e6635a7-7f7a-4363-957e-ea5c0cf20f75.png)

**Comment by MaxNiark** [comment]
dat work rly good ! x) 

![image](https://user-images.githubusercontent.com/126392328/226097459-4243c3fa-41fb-4a41-9428-0474dcfb9e29.png)
